browser/ui/webui/settings/brave_extensions_manifest_v2_installer.cc

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/settings/brave_extensions_manifest_v2_installer.h"
+
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "base/strings/escape.h"
+#include "base/strings/string_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brave/components/constants/brave_services_key.h"
+#include "brave/components/constants/network_constants.h"
+#include "chrome/browser/extensions/crx_installer.h"
+#include "chrome/browser/extensions/extension_install_prompt.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/crx_file/crx_verifier.h"
+#include "components/update_client/update_query_params.h"
+#include "content/public/browser/web_contents.h"
+#include "extensions/browser/crx_file_info.h"
+#include "extensions/common/extension_urls.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "url/gurl.h"
+
+namespace extensions_mv2 {
+
+namespace {
+
+bool IsKnownMV2Extension(std::string_view id) {
+  return id == kNoScriptId || id == kUBlockId || id == kUMatrixId ||
+         id == kAdGuardId;
+}
+
+GURL GetUpdaterExtensionDownloadUrl(std::string_view extenion_id) {
+  const std::string params[] = {base::JoinString({"id", extenion_id}, "="),
+                                "installsource=ondemand", "uc"};
+  const GURL url(
```

Comment:
nit: remove `const` as it prevents rvalue move optimization on return.

Reply:
https://www.cppstories.com/2017/01/const-move-and-rvo,
`Good news is, const doesn’t impact Return Value Optimization (RVO). Compilers are smart enough to handle that`